src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -66,11 +65,10 @@ internal async Task HandleRenameRequest(RenameRequestParams requestParams, Reque
                     requestParams.ConnectionUri,
                     out connInfo))
             {
-                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, ObjectManagementServiceApplicationName))
+                ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementServiceApplicationName);
+                using (serverConnection.SqlConnectionObject)
                 {
-
-                    IRenamable renameObject = this.GetRenamable(requestParams, sqlConn);
-
+                    IRenamable renameObject = this.GetRenamable(requestParams, serverConnection);
```

Comment:
While you're looking at this stuff - right now this is going to throw an ugly error if the SMO object isn't an IRenamable (it'll fail to cast it). I'd hope we don't ever hit that - but just in case it wouldn't hurt to use `as IRenamable` and then if it's null throw a more friendly error.

Reply:
Not a concern as far as ADS is concerned because from the UI side we specify which node types will have the rename support.